ImageD11/sinograms/dataset.py

```diff
@@ -85,7 +96,59 @@ def __repr__(self):
             r.append( '# scans %d from %s to %s'%(
                 len(self.scans), self.scans[0], self.scans[-1] ))
         return  "\n".join(r)
+    
+    def guess_detector(self):
+        '''Guess which detector we are using from the masterfile'''
+        
+        # open the masterfile
+        hname = self.masterfile
+        detectors_seen = []
+        scan = "1.1"
 
+        with h5py.File( hname, 'r' ) as hin:
+            # go through the first scan, and see what detectors we can see
+            for measurement in list(hin[scan]['measurement']):
+                if measurement in POSSIBLE_DETECTOR_NAMES:
+                    detectors_seen.append(measurement)
+        
+        if len(detectors_seen) != 1:
+            raise ValueError("More than one detector seen! Can't work out which one to process.")
+        else:
+            self.detector = detectors_seen[0]
+            
+        
+    def guess_motornames(self):
```

Comment:
To be seen - I think this should come from nexus style attributes or user input to scale in the future